### ISSUE TITLE
Errors in zlib.h

### ISSUE THREAD

**Issue by Robalni** [red-eclipse/base#645]
When I compile using the makefile generated by CMake, I get errors like the one below. I don't get any error when using the other Makefile.
```
include/zlib.h:1583:63: error: conflicting declaration of C function ‘off_t gzseek64(gzFile, off_t, int)’
      ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
                                                               ^
include/zlib.h:1567:30: note: previous declaration ‘off64_t gzseek64(gzFile, off64_t, int)’
    ZEXTERN z_off64_t ZEXPORT gzseek64 OF((gzFile, z_off64_t, int));
                              ^
```

It seems that the reason I only get errors with CMake is that the compile command generated by CMake has -Iinclude but the other one doesn't. This means that when compiling with CMake, it uses the zlib.h and zconf.h inside the include directory in this repository instead of the ones I have installed on my computer. I have tried to replace zlib.h and zconf.h by the ones I have on my computer and that makes it work, so I think we need to upgrade these files to newer versions.

I am using commit 8acae6aa406013441c07315d8bfe10db3858a9ff and OS is Debian 8.6

**Comment by TheAssassin** [comment]
This bug is caused by the for loop that runs for all external dependencies. It runs the `include_directories` macro for zlib, too, but this is actually unnecessary as a header for zlib is distributed within the Red Eclipse source code.
We have to add an exception for zlib in the for loop so that it is skipped in the `include_directories` part.
I will create a branch on my base fork that you can try later.

**Comment by TheAssassin** [comment]
I just tested the build in a fresh Debian 8.6 Docker container and a build with CMake with the existing configuration ([build log](https://paste.kde.org/phqanqpcj/et9t2w)).
Nonetheless I have pushed a proposed fix to my [repository](https://github.com/TheAssassin/redeclipse-base/tree/TheAssassin/issue-645). Please try if that resolves the issue and provide some feedback here.
Side note: In the current Debian stable, [zlib1g](https://packages.debian.org/de/jessie/zlib1g-dev) packages have been updated to use zlib 1.2.8, with some maintainer patches. Ubuntu's [packages](http://packages.ubuntu.com/de/trusty/zlib1g-dev) have used 1.2.8 since trusty, they have some maintainer patches too, but these are different from the ones the Debian maintainers provide. Might be worth investigating the difference.

**Comment by Robalni** [comment]
I still have the same problem when using that proposed fix.

This is a compile command used by the Makefile (without CMake):
```
g++ -fomit-frame-pointer -ffast-math -Wall -fsigned-char -fno-exceptions -fno-rtti -I. -Ishared -Iengine -Igame -Ienet/include -Isupport -I/usr/X11R6/include `sdl2-config --cflags`   -c -o engine/master.o engine/master.cpp
```
and it works. I just need to add `-Iinclude` to get the same error message that I get with CMake:
```
g++ -fomit-frame-pointer -ffast-math -Wall -fsigned-char -fno-exceptions -fno-rtti -I. -Ishared -Iengine -Igame -Ienet/include -Isupport -I/usr/X11R6/include `sdl2-config --cflags` -Iinclude   -c -o engine/master.o engine/master.cpp
```
This one doesn't work. It gives me [this error message](https://paste.debian.net/plainh/8f376c8f).

**Comment by TheAssassin** [comment]
Thinking about it... no wonder this breaks, as the `zlib.h` [header](https://github.com/red-eclipse/base/blob/master/src/include/zlib.h) in the `includes` is for 1.2.5. Looks like this is only important for Windows builds, as a zlib 1.2.5 DLL can be found in the `bin` directory.

You provided the solution with your last comment. Including the `include` directory is obviously redundant, those headers are just relevant when there is no system wide installation. We just have to remove that in the `include_directories` call and it should work. When I created the CMake configuration, it just worked fine on all the systems I tested it on, so I didn't think of that at all.

I'll update my issue branch immediately.
